packages/permission-controller/src/PermissionController.ts

```diff
@@ -1181,6 +1191,67 @@ export class PermissionController<
     this.setCaveat(origin, target, caveatType, caveatValue);
   }
 
+  /**
+   * Updates caveat value across all subjects' permissions that have the specified
+   * target and caveat type.
+   *
+   * @template TargetName - The permission target name. Should be inferred.
+   * @template CaveatType - The valid caveat type for the permission. Should
+   * be inferred.
+   * @param target - The target name of the permission.
+   * @param caveatType - The type of the caveat to update.
+   * @param caveatValue - The new value of the caveat.
+   */
+  updateCaveatForAllSubjects<
+    TargetName extends ExtractPermission<
+      ControllerPermissionSpecification,
+      ControllerCaveatSpecification
+    >['parentCapability'],
+    CaveatType extends ExtractAllowedCaveatTypes<ControllerPermissionSpecification>,
+    CaveatValue extends ExtractCaveatValue<
+      ControllerCaveatSpecification,
+      CaveatType
+    >,
+  >(
+    target: TargetName,
+    caveatType: CaveatType,
+    caveatValue: CaveatValue,
+  ): void {
+    const specification = this.getCaveatSpecification(caveatType);
+    const caveat = { type: caveatType, value: caveatValue };
+
+    this.update((draftState) => {
+      Object.entries(draftState.subjects).forEach(([origin, subject]) => {
+        const { permissions } = subject;
+
+        if (hasProperty(permissions, target)) {
+          const caveatIndex = (
+            permissions[target] as PermissionConstraint
+          ).caveats?.findIndex(
+            (existingCaveat) => existingCaveat.type === caveat.type,
+          );
+          if (caveatIndex !== undefined && caveatIndex !== -1) {
+            if (caveatValue === undefined) {
```

Comment:
What happens if we throw halfway through making the updates? Do all updates get rejected by immer?

Reply:
Maybe it is simpler regardless to throw up front?

Reply:
Yeah...honestly, I moved it into the `forEach` call because the error classes being used in that package require `origin` and I wasn't sure if it was worth it to write out a new error class for this situation because the update would throw on the first iteration regardless if the `caveatValue` is unacceptable.

Reply:
I think its cleaner to throw up front before we start iterating. We can just use `invalidParams`, similarly to here: https://github.com/MetaMask/core/blob/main/packages/permission-controller/src/PermissionController.ts#L1904-L1916

Reply:
Actually, is there a reason we can't just call `validateCaveat`?